File: LeetCode/src/com/cyw/python/Ext_Euclid.py
#扩展欧几里得算法，即辗转相除法
#ax+by=gcb(a, b)
#返回：x,y, gcb(a, b)
#gcb(a, b) ，要不是1，要不是a
def Ext_Euclid ( a , b ):
    if (b == 0):
        return 1 , 0 , a
    else:
        x , y , q=Ext_Euclid( b , a % b )
        x , y = y,( x - (a // b) * y )
        print(q)
        return x , y , q
    
a = Ext_Euclid(47,30)

File: LeetCode/src/com/cyw/algorithms/Ext_Euclid.java
package com.cyw.algorithms;

public class Ext_Euclid {
	
	/**
	 * ·µ»ØµÄx, y, gcb(a, b)
	 * @param a
	 * @param b
	 * @return
	 */
	public static int[] getEx(int a, int b){
		int[] result = new int[3];
		if(b==0){
			result[0] = 1;
			result[1] = 0;
			result[2] = a;
			return result;
		}else{
			result = getEx(b, a%b);
			int t = result[0];
			result[0] = result[1];
			result[1] = t-(a/b)*result[1];
			return result;
		}
	}
	
	public static void main(String [] args){
		int[] result = getEx(17, 12);
		System.out.println(result[0]+" ;"+result[1]+" ;"+result[2]);
//		System.out.println(Math.pow(2,3));
	}

}
